src/shared/guards.ts

```diff
@@ -13,3 +13,7 @@ export const stringGuard = (input: string | undefined): input is string => {
 export const sourceComponentGuard = (input: SourceComponent | undefined): input is SourceComponent => {
   return input instanceof SourceComponent;
 };
+
+export const componentLikeGuard = (input: ComponentLike | undefined): input is ComponentLike => {
+  return input !== undefined && typeof input.fullName === 'string' && typeof input.type === 'string';
```

Comment:
shouldn't `input.fullName` always be a string?
`input.type` can be either a `string` or a `MetadataType` why would we only filter on `string`?

why not do `return input instanceof ComponentLike`, like `sourceComponentGuard`?

Reply:
ComponentLike is an type and not a class that something can be an instanceOf.  So the guard has to validate that it matches the interface (properties and types)

Good catch on the `type` issue--I think I want to use `MetadataMember` exclusively